summoner-cli/src/Summoner/Text.hs

```diff
@@ -26,3 +29,20 @@ headToUpper t = case T.uncons t of
 -- | Convert every element of a list into text, and squash the results
 tconcatMap :: (a -> Text) -> [a] -> Text
 tconcatMap f = T.concat . map f
+
+-- | Aligns a list of texts by their columns
+alignTable :: [[Text]] -> [Text]
+alignTable ghcOutputs = map (T.intercalate " ") $
+    transpose $ zipWith padRow (transpose ghcOutputs) maxWordLengths 
+    where 
+      maxWordLengths :: [Int]
+      maxWordLengths = getMaxLengths ghcOutputs 
```

Comment:
Nice solution with `transpose`! It's quite short and elegant. However, it's not that maintainable in our case type-safety and code readability is more important. I propose to implement `alignTable` according to the following algorithm:

1. Write the following function:
    ```haskell
    mapTriple :: (a -> b) -> (a, a, a) -> (b, b, b)
    ```
2. Calculate maximum length of each component of the triple using `mapTriple`, `foldMap` and `Max + (,,)` monoids.
3. Now you have a list of triples `metas :: [(Text, Text, Text)]` and a triple of maximum lengths of each column: `maxLengths :: (Max Int, Max Int, Max Int)`. You can now write a function: `formatTriple` and do `map (formatTriple maxLengths) metas`

Let me know if you have any other questions :slightly_smiling_face: